browser/brave_ads/application_state/notification_helper/notification_helper_impl_win.cc

```diff
@@ -165,15 +227,14 @@ bool NotificationHelperImplWin::IsFocusAssistEnabled() const {
 
 bool NotificationHelperImplWin::IsNotificationsEnabled() {
   TRACE_EVENT("brave.ads", "NotificationHelperImplWin::IsNotificationsEnabled");
-  HRESULT hr = InitializeToastNotifier();
   auto* toast_notifier = toast_notifier_.Get();
-  if (!toast_notifier || FAILED(hr)) {
+  if (!toast_notifier) {
     VLOG(0) << "Failed to initialize toast notifier";
     return true;
   }
 
   ABI::Windows::UI::Notifications::NotificationSetting notification_setting;
-  hr = toast_notifier->get_Setting(&notification_setting);
+  HRESULT hr = toast_notifier->get_Setting(&notification_setting);
```

Comment:
Does this API and other is fast enough (i.e. can't block it some significant amount of time) to call it from UI thread?
Originally, `InitializeToastNotifier` also looks like this.

Reply:
`InitializeToastNotifier` does several API calls, which affected UI thread performance. The problem was that we called it each time `NotificationHelperImplWin::IsNotificationsEnabled()` was called.
With the current change `InitializeToastNotifier` is executed on background thread only once.
Measuring `NotificationHelperImplWin::IsNotificationsEnabled` trace on my local machine showed that the current change reduced `Wall duration` from 3-4 ms to less then 2ms.

Reply:
Frankly, `IsNotificationsEnabled()` always fit to 4ms for my win-11 machine. But for the perf agent it's not stable. Sometime 4-5ms, sometimes 50-100ms.

The good thing is that we're fixing `InitializeToastNotifier()` that is primary source of issue.
The general issue is that there is no strong guarantee that a random system API doesn't hang for some time for some os version/hardware.

Ok, let's merge it and see how it goes.